services/motion/builtin/builtin.go

```diff
@@ -472,6 +493,21 @@ func (ms *builtIn) planMoveOnMap(
 	if !ok {
 		return nil, nil, fmt.Errorf("cannot move component of type %T because it is not a Base", component)
 	}
+
+	if extra != nil {
+		if profile, ok := extra["motion_profile"]; ok {
+			motionProfile, ok := profile.(string)
+			if !ok {
+				return nil, nil, errors.New("could not interpret motion_profile field as string")
+			}
+			if motionProfile == motionplan.PositionOnlyMotionProfile {
```

Comment:
same thing this can become `kinematicsOptions.PositionOnlyMode = motionProfile == motionplan.PositionOnlyMotionProfile`